File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution 
{
    public int longestConsecutive(int[] nums) 
    {
        if(nums.length < 2)
        {
            return nums.length;
        }
        int len = 1;
        int maxLen = 1;
        Arrays.sort(nums);
        int previous = nums[0];
        for(int i =1; i<nums.length ; i++)
        {
            if(nums[i] == previous)
            {
                continue;
            }
            if(nums[i] == (previous+1))
            {
                len++;
            }
            else
            {
                len = 1;
            }
            previous = nums[i];
            maxLen = Math.max(maxLen, len);

        }
        return maxLen;

    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-4.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:

        nums.sort()
        target = 0

        for num in nums:
            target += num

        if target%2 == 1:
            return False

        target = target/2

        dp = {0}

        for num in nums:
            for i in set(dp):
                if i + num == target:
                    return True
                else:
                    dp.add(i + num)

        return False